File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/Libs/Core/Object/UniqueStrIdGen.cpp
#include "MyProject.h"
#include "UniqueStrIdGen.h"
#include "TypeDef.h"
#include "MClassFactory.h"

MY_BEGIN_NAMESPACE(MyNS)

M_IMPLEMENT_AND_REGISTER_CLASS(UniqueStrIdGen, GObject)

UniqueStrIdGen::UniqueStrIdGen(string prefix, uint baseUniqueId)
	: Super(baseUniqueId)
{
	this->mPrefix = prefix;
}

std::string UniqueStrIdGen::genNewStrId()
{
	this.mRetId = this->mPrefix + "_" + this->genNewId();
	return this->mRetId;
}

std::string UniqueStrIdGen::getCurStrId()
{
	return this->mRetId;
}

std::string UniqueStrIdGen::genStrIdById(uint id)
{
	this.mStringBuilder = new System.Text.StringBuilder();
	this.mStringBuilder.Append(this.mPrefix);
	this.mStringBuilder.Append('_');
	this.mStringBuilder.Append(id);
	this.mRetId = this.mStringBuilder.ToString();
	return this.mRetId;
}

MY_END_NAMESPACE

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/Core/UniqueStrIdGen.cs
namespace SDK.Lib
{
    /**
     * @brief 唯一字符串生成器
     */
    public class UniqueStrIdGen : UniqueNumIdGen
    {
        public const string PlayerPrefix = "PL";
        public const string PlayerChildPrefix = "PC";
        public const string AbandonPlanePrefix = "AP";
        public const string BoomPlanePrefix = "BP";
        public const string RobotPrefix = "RT";
        public const string SnowBlockPrefix = "SM";
        public const string PlayerTargetPrefix = "PT";
        public const string ScoreEffectPrefix = "SE";
        public const string MeteStonePrefix = "MS";
        public const string GroundEntityPrefix = "GE";

        protected string mPrefix;
        protected string mRetId;
        protected System.Text.StringBuilder mStringBuilder;

        public UniqueStrIdGen(string prefix, uint baseUniqueId)
            : base(baseUniqueId)
        {
            this.mPrefix = prefix;
        }

        public string genNewStrId()
        {
            this.mStringBuilder = new System.Text.StringBuilder();
            this.mStringBuilder.Append(this.mPrefix);
            this.mStringBuilder.Append('_');
            this.mStringBuilder.Append(this.genNewId());
            this.mRetId = this.mStringBuilder.ToString();
            return this.mRetId;
        }

        public string getCurStrId()
        {
            return this.mRetId;
        }

        public string genStrIdById(uint id)
        {
            this.mStringBuilder = new System.Text.StringBuilder();
            this.mStringBuilder.Append(this.mPrefix);
            this.mStringBuilder.Append('_');
            this.mStringBuilder.Append(id);
            this.mRetId = this.mStringBuilder.ToString();
            return this.mRetId;
        }
    }
}